pkg/systemd/overview-cards/configurationCard.jsx

```diff
@@ -268,220 +253,175 @@ export class ConfigurationCard extends React.Component {
-            $(note1).text(lengthError);
-            $(note1).css("visibility", "visible");
-            $(note2).css("visibility", "hidden");
-        } else if (validLength && !validName) {
-            $("#sich-hostname-error").addClass("has-error");
-            $(note1).text(charError);
-            $(note1).css("visibility", "visible");
-            $(note2).css("visibility", "hidden");
-        } else {
-            $("#sich-hostname-error").addClass("has-error");
-
-            if ($(note1).text() === lengthError)
-                $(note2).text(charError);
-            else if ($(note1).text() === charError)
-                $(note2).text(lengthError);
-            else {
-                $(note1).text(lengthError);
-                $(note2).text(charError);
-            }
-            $(note1).css("visibility", "visible");
-            $(note2).css("visibility", "visible");
-        }
+    onHostnameChanged(value) {
+        const error = [];
+        if (value.length > 64)
+            error.push(_("Real host name must be 64 characters or less"));
+        if (value.match(/[.a-z0-9-]*/)[0] !== value || value.indexOf("..") !== -1)
+            error.push(_("Real host name can only contain lower-case characters, digits, dashes, and periods (with populated subdomains)"));
+
+        this.setState({
+            hostname: value,
+            update_from_pretty: false,
+            error: error,
+        });
+    }
+
+    onSubmit() {
+        const one = this.state.proxy.call("SetStaticHostname", [this.state.hostname, true]);
+        const two = this.state.proxy.call("SetPrettyHostname", [this.state.pretty, true]);
 
-        apply_button.prop('disabled', !can_apply);
+        Promise.all([one, two]).then(this.props.onClose);
     }
-};
 
-function PageSystemInformationChangeHostname() {
-    this._init();
+    render() {
+        const disabled = this.state.error.length || (this.state.init_hostname == this.state.hostname && this.state.init_pretty == this.state.pretty);
+        return (
+            <Modal isOpen position="top" variant="medium"
+                   onClose={this.props.onClose}
+                   id="system_information_change_hostname"
+                   title={_("Change host name")}
+                   footer={<>
+                       <Button variant='primary' isDisabled={disabled} onClick={this.onSubmit}>{_("Change")}</Button>
+                       <Button variant='link' onClick={this.props.onClose}>{_("Cancel")}</Button>
+                   </>}
+            >
+                <Form isHorizontal>
+                    <FormGroup fieldId="sich-pretty-hostname" label={_("Pretty host name")} hasNoPaddingTop>
+                        <TextInput id="sich-pretty-hostname" value={this.state.pretty} onChange={this.onPrettyChanged} />
+                    </FormGroup>
+                    <FormGroup fieldId="sich-hostname" label={_("Real host name")} hasNoPaddingTop
```

Comment:
Same here.  Otherwise you end up with this:
![Screen Shot 2020-11-17 at 10 23 19](https://user-images.githubusercontent.com/14921356/99371250-f4bf3100-28be-11eb-9411-975e7f43765d.png)